### PR TITLE
Add documentation about kube2iam support

### LINKED ISSUE #253: Add a (optional) feature to enable kube2iam
(Issue itself is work-in-progress, updating it with the parts I found are needed)

In the kube-aws context it seems reasonable to assume that users have IAM enabled, and have some knowledge about it - and that they want it on pod-level, not on worker-level granularity. It would be awesome if kube-aws would provide some OOBE for setting that up using https://github.com/jtblin/kube2iam.

1. Add an "experimental" feature for kube2iam in cluster.yaml
2. If enabled: add this piece to the worker role:
   ```json
           {
            "Action": "sts:AssumeRole",
            "Resource": "*",
            "Effect": "Allow"
        },
   ```
3. If enabled: Deploy the kube2iam daemonset as part of kube-system (or maybe introduce a kube-aws namespace?)
4. Document how to link the worker role to existing roles (roughly like this: 
   ```sh
   EXISTING_ROLES=....
   ROLE_ID=`aws cloudformation list-stack-resources --stack-name CLUSTER-ID | jq -r '.StackResourceSummaries[]|select(.LogicalResourceId=="IAMRoleWorker")|.PhysicalResourceId'`
   ROLE_ARN=`aws iam get-role --role-name collaborne-20170105-IAMRoleWorker-XX0VV92HCQKG | jq -r .Role.Arn`
   for role in ${EXISTING_ROLES}; do
     aws iam get-role --role-name ${role} | jq ".Role.AssumeRolePolicyDocument|(.Statement |= .+ [{\"Principal\":{\"AWS\":\"${ROLE_ARN}\"},\"Action\":\"sts:AssumeRole\",\"Effect\":\"Allow\"}])" > policy-${role}.json
     aws iam update-assume-role-policy --role-name ${role} --policy-document file://$PWD/policy-${role}.json
   done   
   ```

### PR DESCRIPTION
**Pull request by mumoshu**
Closes #253

### CODE REVIEW

File: Documentation/kubernetes-on-aws-add-ons.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ 
+experimental:
+  kube2IamSupport:
+    enabled: true
+
+# for worker nodes
+worker:
+  nodePools:
+  - name: mypool
+    managedIamRoleName: mycontrollerrole
```
Comment (redbaron) [inline Documentation/kubernetes-on-aws-add-ons.md]: shouldn't it be `myworkerrole`?

File: Documentation/kubernetes-on-aws-add-ons.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  Please beware that configuration of target roles' trust relationships are out-of-scope of kube-aws.
+  Please see [the part of kube2iam doc](https://github.com/jtblin/kube2iam#iam-roles) for more information. Basically,
+  you need to point `Principal` to the ARN of a resulting worker/controller IAM role which would look like `arn:aws:iam::<your aws account id>:role/<stack-name>-<managed iam role name>`. 
+
+Finally, an example `cluster.yaml` usable with kube2iam would look like:
+
+```yaml
+# for controller nodes
+controller:
+  managedIamRoleName: myworkerrole
```
Comment (redbaron) [inline Documentation/kubernetes-on-aws-add-ons.md]: `mycontrollerrole`?

**Comment by mumoshu** [comment]
Ah, good catch!
2017年3月23日(木) 22:08 Maxim Ivanov <notifications@github.com>:

**Comment by mumoshu** [comment]
@redbaron Thanks for your review. I've fixed them in https://github.com/kubernetes-incubator/kube-aws/commit/bec0f171a4f8a69b76698e9d46ae53f1706e8242

**Pull request merged**